### DISCUSSION TITLE
Can networkx visualize thousands of nodes ?

### DISCUSSION THREAD

**Discussion by YanqingWu** [networkx/networkx#4715] [Q&A]
I have a graph with thousands of nodes, and I want to visualize it hierarchical like a tree. Can networkx works?

**Comment by dschult** [comment]
It is hard to see what is going on with thousands of nodes.
The software will work. The computer can handle it. But the resulting pictures are often hard to get information out of.
The term "hairball" is often used to describe visualizations that don't convey much information beyond a connected bunch of nodes.

One first attempt in this area would be to use the NetworkX connections to GraphViz through pygraphviz. GraphViz has the program "dot" which does a nice hierarchical layout... 
there are also recent pull requests and issues that talk about and implement hierarchical layouts.

**Reply by YanqingWu** [reply]
I have tried the graphviz, it spent too much time generate the **.pdf** file, so I am seeking some package else.

**Reply by MridulS** [reply]
You could also look at https://gephi.org
